iceoryx_posh/source/mepoo/memory_manager.cpp

```diff
@@ -155,21 +191,25 @@ void MemoryManager::configureMemoryManager(const MePooConfig& f_mePooConfig,
     generateChunkManagementPool(f_managementAllocator);
 }
 
-SharedChunk MemoryManager::getChunk(const MaxSize_t f_size)
+SharedChunk MemoryManager::getChunk(const uint32_t payloadSize,
+                                    const uint32_t payloadAlignment,
+                                    const uint32_t customHeaderSize,
+                                    const uint32_t customHeaderAlignment) noexcept
 {
     void* chunk{nullptr};
     MemPool* memPoolPointer{nullptr};
-    uint32_t adjustedSize = MemoryManager::sizeWithChunkHeaderStruct(f_size);
-    uint32_t totalSizeOfAquiredChunk = 0;
+    uint32_t requiredChunkSize =
+        MemoryManager::requiredChunkSize(payloadSize, payloadAlignment, customHeaderSize, customHeaderAlignment);
+    uint32_t aquiredChunkSize = 0;
 
     for (auto& memPool : m_memPoolVector)
     {
         uint32_t chunkSizeOfMemPool = memPool.getChunkSize();
-        if (chunkSizeOfMemPool >= adjustedSize)
+        if (chunkSizeOfMemPool >= requiredChunkSize)
```

Comment:
It has nothing to do with your pull request but I am realizing that this algorithm makes only then sense when the mempools are ordered from small to large otherwise it can happen that you acquire the 8 Mb chunk for your 10 byte data structure since it is simply the first in the list.

Is this somewhere ensured? Are we testing this?! 

Btw: :point_right: :nose:

Reply:
AFAIK it's checked and not possible to have it ordered otherwise. But I will double check to be sure. It has been a long time since we implemented this.

Reply:
there is a test for this `addMemPoolWrongOrderAtLastElement` and the application is expected to die